hermes-mock/src/main/java/pl/allegro/tech/hermes/mock/HermesMock.java

```diff
@@ -115,33 +161,50 @@ public void expectMessagesOnTopic(int count, String topicName) {
     }
 
     public <T> Optional<T> getLastMessageAs(String topicName, Class<T> clazz) {
-        Optional<LoggedRequest> request = getLastRequest(topicName);
-        if (!request.isPresent()) {
-            return Optional.empty();
-        }
+        return getLastRequest(topicName)
+                .map(req -> deserialize(req, clazz));
+    }
 
-        String bodyAsString = request.get().getBodyAsString();
+    public void resetReceivedRequest() {
+        wireMockServer.resetRequests();
+    }
+
+    public void start() {
+        wireMockServer.start();
+    }
+
+    public void stop() {
+        wireMockServer.stop();
+    }
+
+    private <T> T deserialize(LoggedRequest request, Class<T> clazz) {
         try {
-            return Optional.of(objectMapper.readValue(bodyAsString, clazz));
+            return objectMapper.readValue(request.getBodyAsString(), clazz);
         } catch (IOException ex) {
-            throw new HermesMockException("Cannot read body " + bodyAsString + " as " + clazz.getSimpleName());
+            throw new HermesMockException("Cannot read body " + request.getBodyAsString() + " as " + clazz.getSimpleName());
         }
     }
 
-    public void resetReceivedRequest() {
-        wireMockServer.resetRequests();
+    private byte[] deserializeAvro(LoggedRequest request, Schema schema) {
+        try {
+            return convertToAvro(readJson(request.getBody(), schema), schema);
+        } catch (IOException | AvroRuntimeException ex) {
+            throw new HermesMockException("Failed to convert to AVRO." + ex.getMessage());
```

Comment:
I'm not sure whether we will receive full message with conversion error in message. Maybe we should pass `ex` to `HermesMockException` as `cause` (as a second argument that will be passed to `RuntimeException`)